src/main/java/org/dependencytrack/persistence/FindingsSearchQueryManager.java

```diff
@@ -0,0 +1,374 @@
+            params.put("active", true);
+        }
+        processFilters(filters, queryFilter, params, true);
+        final Query<Object[]> query = pm.newQuery(JDOQuery.SQL_QUERY_LANGUAGE, GroupedFinding.QUERY + queryFilter + (this.orderBy != null ? " ORDER BY " + sortingAttributes.get(this.orderBy) + " " + (this.orderDirection.name().toLowerCase().equals("descending") ? " DESC" : "ASC") : ""));
+        PaginatedResult result = new PaginatedResult();
+        query.setNamedParameters(params);
+        final List<Object[]> totalList = query.executeList();
+        result.setTotal(totalList.size());
+        final List<Object[]> list = totalList.subList(this.pagination.getOffset(), (this.pagination.getOffset() + this.pagination.getLimit() >= totalList.size()) ? totalList.size() : this.pagination.getOffset() + this.pagination.getLimit());
+        final List<GroupedFinding> findings = new ArrayList<>();
+        for (Object[] o : list) {
+            final GroupedFinding finding = new GroupedFinding(o);
+            findings.add(finding);
+        }
+        result.setObjects(findings);
+        return result;
+    }
+
+    private void processFilters(Map<String, String> filters, StringBuilder queryFilter, Map<String, Object> params, boolean isGroupedByVulnerabilities) {
+        for (String filter : filters.keySet()) {
+            switch (filter) {
+                case "severity" -> processArrayFilter(queryFilter, params, filter, filters.get(filter), "\"VULNERABILITY\".\"SEVERITY\"");
+                case "analysisStatus" -> processArrayFilter(queryFilter, params, filter, filters.get(filter), "\"ANALYSIS\".\"STATE\"");
+                case "vendorResponse" -> processArrayFilter(queryFilter, params, filter, filters.get(filter), "\"ANALYSIS\".\"RESPONSE\"");
+                case "publishDateFrom" -> processRangeFilter(queryFilter, params, filter, filters.get(filter), "\"VULNERABILITY\".\"PUBLISHED\"", true, true, false);
+                case "publishDateTo" -> processRangeFilter(queryFilter, params, filter, filters.get(filter), "\"VULNERABILITY\".\"PUBLISHED\"", false, true, false);
+                case "attributedOnDateFrom" -> processRangeFilter(queryFilter, params, filter, filters.get(filter), "\"FINDINGATTRIBUTION\".\"ATTRIBUTED_ON\"", true, true, false);
+                case "attributedOnDateTo" -> processRangeFilter(queryFilter, params, filter, filters.get(filter), "\"FINDINGATTRIBUTION\".\"ATTRIBUTED_ON\"", false, true, false);
+                case "textSearchField" -> processInputFilter(queryFilter, params, filter, filters.get(filter), filters.get("textSearchInput"));
+                case "cvssFrom" -> processRangeFilter(queryFilter, params, filter, filters.get(filter), "\"VULNERABILITY\".\"CVSSV3BASESCORE\"", true, false, false);
+                case "cvssTo" -> processRangeFilter(queryFilter, params,filter, filters.get(filter), "\"VULNERABILITY\".\"CVSSV3BASESCORE\"", false, false, false);
```

Comment:
We'll need equivalent filters for CVSSv2, and in the near future also CVSSv4. Naming the filter (and input fields in the frontend) `cvssFrom` and `cvssTo` could be a bit confusing.